Cbermudez97 t6 vote command

cbermudez97 opened the pull request:
Fixes #6 #7 and #8

cbermudez97 committed:
[bot] refs #6 - Implement helper methods for vote command

cbermudez97 committed:
[bot] refs #6 - Create filters.py to hold common used filters

cbermudez97 committed:
[bot] refs #6 - Implement base conversational vote handler

cbermudez97 committed:
Merge remote-tracking branch 'origin/main' into cbermudez97_t6_vote_command

cbermudez97 committed:
[bot] refs #6 - Create vote register command handler

cbermudez97 committed:
[bot] refs #6 - Fix errors in vote register handler

cbermudez97 committed:
[bot] refs #6 - Implement vote command handler using callbacks

cbermudez97 committed:
[bot] refs #6 - Add control for cases when the voting callbacks are too
old

cbermudez97 committed:
[bot] refs #6 - Fix typos in responses

e1Ru1o reviewed (commented):
Move `parse_cdata` and `build_cdata` to utils.py

cbermudez97 committed:
[bot] refs #6 - Fix invalid message asking for a discussion with no valid ones available

cbermudez97 committed:
[box] refs #6 - Move helpers methods to utils.py

cbermudez97 committed:
[bot] refs #6 - Create method to parse selected voting options from message

cbermudez97 committed:
[bot] refs #6 - Correctly import utils methods

cbermudez97 committed:
[bot] refs #6 - Change purpose of the voting_in dict in user data

voting_in is now a set that holds ids of chats were the user registered into a discussion. The selected options are parsed from the message text that list them.

cbermudez97 committed:
[bot] refs #6 - Fix errors when confirming a vote

cbermudez97 committed:
[bot] refs #7 - Rename vote callback helpers methods

cbermudez97 committed:
[bot] refs #7 - Move vote callback helpers to vote.py

cbermudez97 committed:
[bot] refs #8 - Create command handler to list voters in a group

cbermudez97 committed:
[bot] refs #8 - Add list command handler to bot handlers
